fdbclient/NativeAPI.actor.cpp

```diff
@@ -8124,6 +8157,20 @@ Future<Void> DatabaseContext::createSnapshot(StringRef uid, StringRef snapshot_c
 	return createSnapshotActor(this, UID::fromString(uid_str), snapshot_command);
 }
 
+Future<DatabaseSharedState*> DatabaseContext::initSharedState() {
+	ASSERT(!sharedStatePtr); // Don't re-initialize shared state if a pointer already exists
+	DatabaseSharedState* newState = new DatabaseSharedState();
+	setSharedState(newState);
+	return newState;
+}
+
+Future<Void> DatabaseContext::setSharedState(DatabaseSharedState* p) {
+	MutexHolder holder(p->mutexLock);
```

Comment:
Is the intent of this mutex to guard the ref count increment? Possible we could avoid that if we made ref count an atomic.